R/query.R

```diff
@@ -284,6 +284,22 @@ is_id_lookup <- function(expr) {
   expr$type == "lookup" && expr$name == "id"
 }
 
+is_logical <- function(expr) {
+  truthy <- c(quote(TRUE), quote(FALSE), quote(true), quote(false))
+  bool <- vlapply(truthy, function(x) x == expr)
+  sum(bool) == 1
```

Comment:
```suggestion
  truthy <- list(TRUE, FALSE, quote(true), quote(false))
  any(vlapply(truthy, identical))
```

I feel like use of `==` here might do coersions. You could also do this as:

```
(is.logical(expr) && !is.na(expr)) || (is.symbol(expr) && as.character(expr) %in% c("true", "false"))
```